src/api.ts

```diff
@@ -93,16 +93,23 @@ router.post(
     const dto: BidsListDto = req.body as BidsListDto;
     const nftBids: BidsList = {};
 
-    // Get all of the bids on every provided nftId
+    // Create an empty array for each given nftId
     for (const nftId of dto.nftIds) {
-      try {
-        const bids = await database.getBidsByNftId(nftId);
-        nftBids[nftId] = bids;
-      } catch {
-        // Returns 500
-        throw Error(`Failed to list bids for NFT: ${nftId}`);
+      nftBids[nftId] = [];
+    }
+
+    try {
+      const bids = await database.getBidsByNftIds(dto.nftIds);
+
+      // For each bid, map to appropriate nftId array
+      for (const bid of bids) {
+        const nftId = bid.nftId;
+        nftBids[nftId]?.push(bid);
       }
+    } catch {
+      throw Error("Could not get bids for given nftIds");
     }
+
     return res.status(200).send(nftBids);
```

Comment:
might be worth returning a 404 if the list of bids comes back empty from the db

Reply:
I would say a 404 is "we could not find that nftId" because an empty bid array was still successful in finding that bid array, it just happens to have no information